packages/rulesets/src/oas/index.ts

```diff
@@ -33,7 +33,60 @@ const ruleset = {
   formats: [oas2, oas3, oas3_0, oas3_1],
   aliases: {
     PathItem: '$.paths[*]',
+    ResponsesObject: {
+      targets: [
+        { formats: [oas2], given: ['$.responses', '#OperationObject.responses'] },
+        { formats: [oas3], given: ['$.components.responses', '#OperationObject.responses'] },
+      ],
+    },
+    ResponseObject: '#ResponsesObject[*]', // [(@ && !@.$ref)]
+    ParametersDefinitionsObject: {
+      targets: [
+        { formats: [oas2], given: ['$.parameters'] },
+        { formats: [oas3], given: ['$.components.parameters'] },
+      ],
+    },
+    ParametersObject: {
+      targets: [
+        { formats: [oas2], given: ['#PathItem.parameters', '#OperationObject.parameters'] },
+        {
+          formats: [oas3],
+          given: ['#PathItem.parameters', '#OperationObject.parameters'],
+        },
+      ],
+    },
+    ParameterObject: {
+      targets: [
+        { formats: [oas2], given: ['#ParametersDefinitionsObject[*]', '#ParametersObject[*]'] }, // ?(@ && !@.$ref)
+        {
+          formats: [oas3],
+          given: ['#ParametersDefinitionsObject[*]', '#ParametersObject[*]'], // [?(@ && !@.$ref)]
+        },
+      ],
+    },
+    EncodingObject: '#MediaTypeObject.encoding',
+    MediaTypeObject: ['#RequestBodyObject.content', '#ParameterObject.content', '#ResponseObject.content'],
+    HeadersObject: {
+      targets: [
+        { formats: [oas2], given: ['#ResponseObject.headers'] },
+        {
+          formats: [oas3],
+          given: ['$.components.headers', '#EncodingObject.headers', '#ResponseObject.headers'],
+        },
+      ],
+    },
+    HeaderObject: {
+      targets: [
+        { formats: [oas2], given: ['#ResponseObject.headers[*]'] },
+        {
+          formats: [oas3],
+          given: ['#HeadersObject[*]'], // ?(@ && !$.ref)
+        },
+      ],
+    },
+    RequestBodyObject: ['#OperationObject.requestBody', '$.components.requestBodies[*]'],
     OperationObject: '#PathItem[get,put,post,delete,options,head,patch,trace]',
+    SchemaObject: ['#MediaTypeObject.schema', '#ParameterObject.schema', '$.components.schemas[*]'],
```

Comment:
Isn't this a little misleading? I mean, isn't there a possibility of a lot of schema objects embedded in properties (and embedded in properties of properties, and so on) that won't be captured here?

Reply:
Yeah, it's not final yet. `SchemaObject` will have to catch all schemas, I've got it somewhere but not committed yet